p2p/server.go

```diff
@@ -196,15 +196,18 @@ func (serv *ExchangeServer[H]) handleRequestByHash(hash []byte) ([]H, error) {
 	return []H{h}, nil
 }
 
-// handleRequest fetches the Header at the given origin and
+// handleRangeRequest fetches the Header at the given origin and
 // writes it to the stream.
-func (serv *ExchangeServer[H]) handleRequest(from, to uint64) ([]H, error) {
+func (serv *ExchangeServer[H]) handleRangeRequest(from, to uint64) ([]H, error) {
 	if from == uint64(0) {
 		return serv.handleHeadRequest()
 	}
 
 	startTime := time.Now()
-	ctx, span := tracerServ.Start(serv.ctx, "request-range", trace.WithAttributes(
+	log.Debugw("server: handling range request", "from", from, "to", to)
+	ctx, cancel := context.WithTimeout(serv.ctx, serv.Params.RangeRequestTimeout)
+	defer cancel()
+	ctx, span := tracerServ.Start(ctx, "request-range", trace.WithAttributes(
```

Comment:
It looks like RangeRequestTimeout is also set in handleRequestByHash. It would be better to apply the timeout universally across all handlers. We can do this by:

Changing RangeRequestTimeout to RequestTimeout.
Moving the context construction up to the requestHandler function in the router and passing the context with the timeout to the handlers.
This ensures that all handlers always have a timeout, making it cleaner and more consistent.

Reply:
Good points. Renaming will make usage of this param consistent.

Done. PR became bigger that I initially planned but the change is simpler. The only one drawback is that we've a breaking change due to renamed param. Gonna update PR.